tests/components/alarm_control_panel/test_init.py

```diff
@@ -0,0 +1,13 @@
+"""Tests for Alarm control panel."""
+from homeassistant.components import alarm_control_panel
+
+
+def test_deprecated_base_class(caplog):
+    """Test deprecated base class."""
+
+    class CustomAlarm(alarm_control_panel.AlarmControlPanel):
+        def supported_features(self):
+            pass
+
+    CustomAlarm()
+    assert "AlarmControlPanel is deprecated, modify CustomAlarm" in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5418277)